src/Microsoft.Azure.CosmosEventSourcing/Stores/DefaultEventStore.Def.cs

```diff
@@ -3,14 +3,18 @@
 
 using System.Runtime.CompilerServices.Context;
 using Microsoft.Azure.CosmosEventSourcing.Items;
+using Microsoft.Azure.CosmosEventSourcing.Options;
 using Microsoft.Azure.CosmosRepository;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.CosmosEventSourcing.Stores;
 
 internal partial class DefaultEventStore<TEventItem>(
     IBatchRepository<TEventItem> batchRepository,
     IReadOnlyRepository<TEventItem> readOnlyRepository,
-    IContextService contextService) :
+    IContextService contextService,
+    IOptionsMonitor<CosmosEventSourcingOptions> optionsMonitor) :
     IEventStore<TEventItem> where TEventItem : EventItem
 {
+    private readonly IOptionsMonitor<CosmosEventSourcingOptions> _optionsMonitor = optionsMonitor;
```

Comment:
This field isn't needed, you can access the `optionsMonitor` anywhere within the `partial class`. These are primary constructors.

Reply:
ahhh that makes sense now, I'll update this cheers!